sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Product.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions.aggregate
+
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.catalyst.dsl.expressions._
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, Expression, ImplicitCastInputTypes, Literal}
+import org.apache.spark.sql.catalyst.util.TypeUtils
+import org.apache.spark.sql.types.{AbstractDataType, DataType, DoubleType}
+
+
+/** Multiply numerical values within an aggregation group */
+case class Product(child: Expression, scale: Double = 1.0)
```

Comment:
Sorry if I just missed this in the recent comments, but this still has a scale param - can be removed?

Reply:
I'm more inclined to leave this - it's an internal Catalyst mechanism that seems like a useful place to allow optimizations associated with different scale values.

Reply:
It can be added back later if needed? this adds a tiny bit of overhead in the processing

Reply:
Yeah, I think we can just remove `scale` here, and define it as a `private val`

Reply:
Yeah, I think we can just remove `scale` here. It's always 1.0, right.

Reply:
Ok - I've removed "scale" completely now.